helper-cli/src/main/kotlin/commands/CreateAnalyzerResultCommand.kt

```diff
@@ -115,15 +120,20 @@ private data class ScannedPackage(
     val scanTime: Instant
 )
 
-private fun getScannedPackages(connection: Connection, ids: Collection<Identifier>): List<ScannedPackage> {
+private fun getScannedPackages(
+    connection: Connection,
+    ids: Collection<Identifier>,
+    scanCodeVersion: String?
+): List<ScannedPackage> {
     val query = """
         SELECT 
             s.identifier as id,
             s.scan_result->>'provenance' as provenance,
             (s.scan_result->'summary'->>'start_time')::timestamp as start_time
         FROM 
             scan_results s 
-        WHERE 
+        WHERE
+            ${ scanCodeVersion?.let { "s.scan_result->'scanner'->>'version' = '$it' AND" }.orEmpty() }
```

Comment:
Just an idea (typing blindly):

    val whereClause = listOfNotNull(
        scanCodeVersion?.let { "s.scan_result->'scanner'->>'version' = '$it'",
        "s.identifier = ANY(?)"
    ).joinToString(" AND ")

    // ....

    WHERE $whereClause;